internal/runtimehandlerhooks/high_performance_hooks.go

```diff
@@ -356,6 +420,175 @@ func setCPUQuota(cpuMountPoint, parentDir string, c *oci.Container, enable bool)
 	return nil
 }
 
+// setCPUPMQOSResumeLatency sets the pm_qos_resume_latency_us for a cpu and stores the original
+// value so it can be restored later. If the latency is an empty string, the original latency
+// value is restored.
+// Note: Passing the directories as parameters to facilitate unit tests (following the
+// pattern used by other functions in this file).
+func setCPUPMQOSResumeLatency(c *oci.Container, latency, cpuDir, cpuSaveDir string) error {
+	lspec := c.Spec().Linux
+	if lspec == nil ||
+		lspec.Resources == nil ||
+		lspec.Resources.CPU == nil ||
+		lspec.Resources.CPU.Cpus == "" {
+		return errors.Errorf("find container %s CPUs", c.ID())
+	}
+
+	cpus, err := cpuset.Parse(lspec.Resources.CPU.Cpus)
+	if err != nil {
+		return err
+	}
+
+	for _, cpu := range cpus.ToSlice() {
+		if latency != "" {
+			// retrieve the current latency
+			latencyFile := fmt.Sprintf("%s/cpu%d/power/pm_qos_resume_latency_us", cpuDir, cpu)
+			latencyOrig, err := os.ReadFile(latencyFile)
+			if err != nil {
+				return err
+			}
+
+			// save the current latency so we can restore it later
+			cpuPowerSaveDir := fmt.Sprintf("%s/cpu%d/power", cpuSaveDir, cpu)
+			err = os.MkdirAll(cpuPowerSaveDir, 0o750)
+			if err != nil {
+				return err
+			}
+			latencyFileOrig := path.Join(cpuPowerSaveDir, "pm_qos_resume_latency_us")
+			err = os.WriteFile(latencyFileOrig, latencyOrig, 0o644)
+			if err != nil {
+				return err
+			}
+
+			// update the pm_qos_resume_latency_us
+			err = os.WriteFile(latencyFile, []byte(latency), 0o644)
+			if err != nil {
+				return err
```

Comment:
Question: if we were unable to set the new latency, should we not be saving the original one (as there's no need to restore it)?

Reply:
It should be an extremely rare case where we couldn't set the new latency. The only improvement here would be to delete the saved latency if we couldn't set the new latency - I still want to save the original latency first, as I wouldn't want to change the latency if I knew I wasn't going to be able to restore it to the original value. Since this failure case is extremely unlikely and there is no adverse impact of attempting to restore the original (unchanged) latency, I'd prefer to keep the code simple and not add in the extra error handling.